src/api/bulk2.ts

```diff
@@ -355,8 +356,7 @@ export class QueryJobV2<S extends Schema> extends EventEmitter {
   private jobInfo?: QueryJobInfoV2;
   private locator?: string;
 
-  constructor(conn: Connection<S>, options: ExistingQueryJobV2Options);
-  constructor(conn: Connection<S>, options: CreateQueryJobV2Options);
+  constructor(conn: Connection<S>, options: ExistingQueryJobV2Options | CreateQueryJobV2Options);
```

Comment:
nice catch, it seems even this line can be removed since the constructor below has the same signature.